http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ApacheHttpClient.java

```diff
@@ -154,9 +156,16 @@ private ConnectionManagerAwareHttpClient createClient(ApacheHttpClient.DefaultBu
     }
 
     private void addProxyConfig(HttpClientBuilder builder,
-                                ProxyConfiguration proxyConfiguration) {
-        if (isProxyEnabled(proxyConfiguration)) {
+                                DefaultBuilder configuration) {
+        ProxyConfiguration proxyConfiguration = configuration.proxyConfiguration;
 
+        if (configuration.httpRoutePlanner != null) {
```

Comment:
I wonder if we should validate the builder to make sure httpRoutePlanner and proxyConfigurations are not present at the same time and throw exception if they are like what we do in netty client.

https://github.com/aws/aws-sdk-java-v2/blob/0a838db69b0f1e08e2a5b9b3abf28036cb6b2a5a/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java#L137-L138

Reply:
Will do, also shouldn't that error message say eventLoopGroupBuilder and not eventLoopGroupFactory?